glue/sales_val_flagging.py

```diff
@@ -152,30 +156,115 @@ def group_size_adjustment(df, stat_groups: list, min_threshold, condos: bool):
         df: dataframe with newly manually adjusted outlier values.
 
     """
+
     group_counts = df.groupby(stat_groups).size().reset_index(name="count")
     filtered_groups = group_counts[group_counts["count"] <= min_threshold]
 
     # Merge df_flagged with filtered_groups on the columns to get the matching rows
-    merged_df = pd.merge(
+    df = pd.merge(
         df, filtered_groups[stat_groups], on=stat_groups, how="left", indicator=True
     )
 
-    # Modify the .loc condition to include checking for sv_outlier_type values
-    condition = (merged_df["_merge"] == "both") & (merged_df["sv_is_outlier"] == 1)
+    # Assign sv_outlier_reasons
+    for idx in range(1, 4):
+        df[f"sv_outlier_reason{idx}"] = np.nan
+
+    outlier_type_crosswalk = {
+        "sv_ind_price_high_price": "High price",
+        "sv_ind_ptax_flag_w_high_price": "High price",
+        "sv_ind_price_low_price": "Low price",
+        "sv_ind_ptax_flag_w_low_price": "Low price",
+        "sv_ind_price_high_price_sqft": "High price per square foot",
+        "sv_ind_ptax_flag_w_high_price_sqft": "High price per square foot",
+        "sv_ind_price_low_price_sqft": "Low price per square foot",
+        "sv_ind_ptax_flag_w_low_price_sqft": "Low price per square foot",
+        "sv_ind_ptax_flag": "PTAX-203 Exclusion",
+        "sv_ind_char_short_term_owner": "Short-term owner",
+        "sv_ind_char_family_sale": "Family Sale",
+        "sv_ind_char_non_person_sale": "Non-person sale",
+        "sv_ind_char_statistical_anomaly": "Statistical Anomaly",
+        "sv_ind_char_price_swing_homeflip": "Price swing / Home flip",
+    }
+
+    """
+    During our statistical flagging process, we automatically discard
+    a sale's eligibility for outlier status if the number of sales in 
+    the statistical grouping is below a certain threshold. The list - 
+    `group_thresh_price_fix` along with the ['_merge'] column will allow
+    us to exclude these sales for the sv_is_outlier status.
+
+    Since the `sv_is_outlier` column requires a price value, we simply
+    do not assign these price outlier flags if the group number is below a certain
+    threshold
+
+    Note: This doesn't apply for sales that also have a ptax outlier status.
+          In this case, we still assign the price outlier status.
+    """
+    group_thresh_price_fix = [
+        "sv_ind_price_high_price",
+        "sv_ind_price_low_price",
+        "sv_ind_price_high_price_sqft",
+        "sv_ind_price_low_price_sqft",
+    ]
 
-    # Using .loc[] to set the desired values for rows meeting the condition
-    merged_df.loc[condition, "sv_outlier_type"] = "Not outlier"
-    merged_df.loc[condition, "sv_is_outlier"] = 0
+    def fill_outlier_reasons(row):
+        reason_idx = 1
```

Comment:
**[Nitpick, non-blocking]** Now that we're keeping track of the set of `reasons_added`, I don't think we need this counter anymore -- we can just replace it with references to `len(reasons_added)`. Note that we'd need to be careful with off-by-one errors the context of the `outlier_types_crosswalk` loop, since we want to set the column name to `sv_outlier_reason{len(reasons_added) + 1}` and then subsequently check `if len(reasons_added) >= 3`. (We could also consider just setting `reason_idx = len(reasons_added) + 1` before setting `sv_outlier_reason{reason_idx}` in the loop if the `if` conditional passes.)